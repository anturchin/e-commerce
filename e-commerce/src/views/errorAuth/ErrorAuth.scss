.overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(126, 126, 126, 0.5);
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 9999;
    &.hidden {
        display: none;
    }
}
