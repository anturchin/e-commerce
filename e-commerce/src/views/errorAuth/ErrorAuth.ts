import { View } from '../View';
import { ErrorButton } from './errorButton/ErrorButton';
import { ErrorMessage } from './errorMessage/ErrorMessage';
import { ErrorWrapper } from './errorWrapper/ErrorWrapper';

import './ErrorAuth.scss';

export class ErrorAuth extends View {
    private errorMessage: ErrorMessage;

    private errorWrapper: ErrorWrapper;

    private errorButton: ErrorButton;

    constructor() {
        super({ tag: 'div', classNames: ['overlay', 'hidden'] });

        this.errorMessage = new ErrorMessage();
        this.errorWrapper = new ErrorWrapper();
        this.errorButton = new ErrorButton();

        this.setupErrorAuth();
        this.setupEvent();
    }

    public showMessage(message: string, status: number): void {
        this.errorMessage.getElement().textContent = ` [ ${status} ]: ${message}`;
        this.getElement().classList.remove('hidden');
    }

    public hideMessage(): void {
        this.getElement().classList.add('hidden');
    }

    private setupEvent(): void {
        this.errorButton.getElement().addEventListener('click', () => this.hideMessage());
    }

    private setupErrorAuth(): void {
        this.errorWrapper
            .getElement()
            .append(this.errorMessage.getElement(), this.errorButton.getElement());
        this.addInnerElement(this.errorWrapper.getElement());
    }
}
