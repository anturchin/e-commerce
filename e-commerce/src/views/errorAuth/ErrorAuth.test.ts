import { describe, expect, test } from '@jest/globals';
import { ErrorAuth } from './ErrorAuth';

describe('ErrorAuth', () => {
    test('should match', () => {
        const errorAuth = new ErrorAuth();
        const renderedErrorAuth = errorAuth.getElement();
        expect(renderedErrorAuth.outerHTML).toMatchSnapshot();
    });
});
