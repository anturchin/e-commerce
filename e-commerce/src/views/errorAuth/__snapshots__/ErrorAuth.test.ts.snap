// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAuth should match 1`] = `"<div class="overlay hidden"><div class="modal-error"><p class="error__text"></p><button class="error__close">close</button></div></div>"`;
